packages/toolpad-app/pages/_app.tsx

```diff
@@ -1,27 +1,46 @@
 import * as React from 'react';
 import Head from 'next/head';
-import App, { AppContext, AppProps } from 'next/app';
+import App, { AppContext, AppProps, NextWebVitalsMetric } from 'next/app';
 import { ThemeProvider } from '@mui/material/styles';
 import CssBaseline from '@mui/material/CssBaseline';
 import { CacheProvider, EmotionCache } from '@emotion/react';
 import { QueryClientProvider } from '@tanstack/react-query';
 import { LicenseInfo } from '@mui/x-data-grid-pro';
+import { useRouter } from 'next/router';
 import theme from '../src/theme';
 import createEmotionCache from '../src/createEmotionCache';
 import { MUI_X_PRO_LICENSE } from '../src/constants';
 import { queryClient } from '../src/api';
+import { reportWebVitalsToGA, setGAPage } from '../src/utils/ga';
 
 LicenseInfo.setLicenseKey(MUI_X_PRO_LICENSE);
 
 // Client-side cache, shared for the whole session of the user in the browser.
 const clientSideEmotionCache = createEmotionCache();
 
+export const reportWebVitals = (metric: NextWebVitalsMetric): void => {
+  reportWebVitalsToGA(metric);
+};
+
 interface MyAppProps extends AppProps {
   emotionCache?: EmotionCache;
 }
 
 export default function MyApp(props: MyAppProps) {
   const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;
+
+  const router = useRouter();
+
+  React.useEffect(() => {
```

Comment:
@apedroferreira 
- How useful would this really be? We also have `react-router` on the editor, so the data wouldn't be exactly a good representation of user navigation throughout the app.
- Did you verify we only send one event on hard reload?

Reply:
I took it from this example https://github.com/vercel/next.js/tree/canary/examples/with-google-analytics, just rechecked again and as far as I can tell:
- hard reload only triggers one page view
- client site router changes (creating a new connection for example) trigger page views too, probably because of `hashChangeComplete`

Reply:
seems to run for the iframe too, not sure if that's helpful

Reply:
might make sense to disable for the app canvas

Reply:
ok actually it turns out this effect is doing nothing at all... the page view events are being sent automatically.
i'll remove this effect and see if i can filter out events from the iframe

Reply:
and because the page view events are sent automatically doesn't seem simple to filter out the ones from the canvas... - we can just ignore that data in the google analytics dashboard itself if we want to use page view data, im gonna make a PR that just removes the effect